src/main/java/com/marketdata/MarketSimulatorBackend/api/controller/StockController.java

```diff
@@ -13,19 +15,18 @@
 @RestController
 @RequestMapping("/api")
 public class StockController {
-
-    private StockService stockService;
+    private final StockServiceProvider stockServiceProvider;
```

Comment:
The word provider is misleading, I think what you mean is service.
The application should have three main layers:
1. Presentation layer: Where you have your controllers. This should be very thin layer with no logic in it at all.
2. Service layer: Where you have the services that the controllers will call.
3. Infra layer: where you have providers (the word provider is used for clients that connect directly with 3rd party services for example), backend clients, configs, db connections, storage layer, ....etc